File: app/src/main/java/de/karzek/diettracker/presentation/main/cookbook/recipeManipulation/dialog/editMeals/adapter/viewHolder/MealViewHolder.java
package de.karzek.diettracker.presentation.main.cookbook.recipeManipulation.dialog.editMeals.adapter.viewHolder;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.widget.CheckBox;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import de.karzek.diettracker.R;
import de.karzek.diettracker.presentation.model.MealDisplayModel;

/**
 * Created by MarjanaKarzek on 30.05.2018.
 *
 * @author Marjana Karzek
 * @version 1.0
 * @date 30.05.2018
 */
public class MealViewHolder extends RecyclerView.ViewHolder {

    @BindView(R.id.checkbox) CheckBox meal;

    private final OnItemCheckedChangeListener onItemCheckedChangeListener;

    public MealViewHolder(ViewGroup viewGroup,
                          OnItemCheckedChangeListener onItemCheckedChangeListener) {
        super(LayoutInflater.from(viewGroup.getContext())
                .inflate(R.layout.item_dialog_checkbox, viewGroup, false));
        ButterKnife.bind(this, itemView);
        this.onItemCheckedChangeListener = onItemCheckedChangeListener;
    }

    public void bind(MealDisplayModel displayModel, boolean status) {
        itemView.setTag(displayModel.getId());

        meal.setText(displayModel.getName());
        meal.setChecked(status);
    }

    @OnClick (R.id.checkbox) public void onCheckedChanged(){
        onItemCheckedChangeListener.onItemCheckChanged((int) itemView.getTag(), meal.isChecked());
    }

    public interface OnItemCheckedChangeListener {
        void onItemCheckChanged(int id, boolean checked);
    }

}

File: app/src/main/java/de/karzek/diettracker/presentation/main/cookbook/recipeManipulation/dialog/editMeals/adapter/viewHolder/MealViewHolder.kt
package de.karzek.diettracker.presentation.main.cookbook.recipeManipulation.dialog.editMeals.adapter.viewHolder

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import de.karzek.diettracker.presentation.model.MealDisplayModel
import kotlinx.android.extensions.LayoutContainer
import kotlinx.android.synthetic.main.item_dialog_checkbox.*

/**
 * Created by MarjanaKarzek on 30.05.2018.
 *
 * @author Marjana Karzek
 * @version 1.0
 * @date 30.05.2018
 */
class MealViewHolder(override val containerView: View,
                     private val onItemCheckedChangeListener: OnItemCheckedChangeListener
) : RecyclerView.ViewHolder(containerView), LayoutContainer {

    fun bind(displayModel: MealDisplayModel, status: Boolean) {
        itemView.tag = displayModel.id

        checkbox.text = displayModel.name
        checkbox.isChecked = status

        checkbox.setOnClickListener { onItemCheckedChangeListener.onItemCheckChanged(itemView.tag as Int, checkbox.isChecked) }
    }

    interface OnItemCheckedChangeListener {
        fun onItemCheckChanged(id: Int, checked: Boolean)
    }

}
